libs/core/src/operators/substream.pipe.ts

```diff
@@ -0,0 +1,106 @@
+import { Action } from '@ngrx/store';
+import { pipe, Observable, MonoTypeOperatorFunction, Operator, TeardownLogic, Subscriber } from 'rxjs';
+import { scan } from 'rxjs/operators';
+
+class SkipWhileSubscriber<T> extends Subscriber<T> {
+  private skipping = true;
+  private index = 0;
+
+  constructor(
+    destination: Subscriber<T>,
+    private predicate: (value: T, index: number) => boolean
+  ) {
+    super(destination);
+  }
+
+  protected _next(value: T): void {
+    const destination = this.destination;
+
+    // call the predicate every time
+    // this is the only change from the vanilla rxjs operator
+    this.tryCallPredicate(value);
+
+    if (!this.skipping) {
+      destination.next(value);
+    }
+  }
+
+  private tryCallPredicate(value: T): void {
+    try {
+      const result = this.predicate(value, this.index++);
+      this.skipping = Boolean(result);
+    } catch (err) {
+      this.destination.error(err);
+    }
+  }
+}
+
+class SkipWhileOperator<T> implements Operator<T, T> {
+  constructor(private predicate: (value: T, index: number) => boolean) {
+  }
+
+  call(subscriber: Subscriber<T>, source: any): TeardownLogic {
+    return source.subscribe(new SkipWhileSubscriber(subscriber, this.predicate));
+  }
+}
+
+/**
+ * All the skipWhile stuff is copied directly from rxjs/src/internal/operators/skipWhile.ts.
+ * The only change is that it will continue to check the predicate even while not skipping.
+ */
+function skipWhile<T>(predicate: (value: T, index: number) => boolean): MonoTypeOperatorFunction<T> {
+  return (source: Observable<T>) => source.lift(new SkipWhileOperator(predicate));
+}
+
+export type ActionType = Action['type'];
+export type ActionSequenceStep = ActionType | ActionType[];
+export type ActionSequence = ActionSequenceStep[];
+
+const compareActionSequenceStep = (step: ActionSequenceStep, action: Action): boolean => Array.isArray(step)
+  ? step.reduce((acc, type) => acc || action.type === type, false)
+  : action.type === step
+
+/**
+ * Finds and emits a substream of actions.
```

Comment:
I think this should be something like: "Watches for a particular stream of actions, and emits a list of those actions once all actions have occurred"

Reply:
That's better. I think `Watches for a particular stream of actions in the specified order and emits a list of those actions once all sequence steps have been matched.` is even a little more precise.